### DISCUSSION TITLE
implementing a custom decorator which does not return a value

### DISCUSSION THREAD

**Discussion by emadbaqeri** [MichalLytek/type-graphql#1265] [Q&A] [answered]
Hey folks, I'm wondering how can I implement a decorator that is handling the authorization part of the application without returning any value inside the decorator. First I started to use the `AuthChecker` function and implemented a `customeAuthChecker` exact the same as the documentation. The problem is I need to pass an argument to the auth checker like the `ID` of some product to validate if the user has access to the product or not. 
Can you please help me on this, any ideas or suggestions would be appreciated ;)

**Comment by emadbaqeri** [comment]
I have hopefully found a way ;)

**Comment by MichalLytek** [comment, marked as answer]
Parameter decorator is for returning values. Method decorator is for running action before/after the resolver via the middlewares.

**Reply by emadbaqeri** [reply]
> Parameter decorator is for returning values. Method decorator is for running action before/after the resolver via the middlewares.

Yeah I see, thanks for response. The problem was implementing a dynamic authorized function and `authChecker` helper for a multi layer access control system.

**Reply by MichalLytek** [reply]
In some discussion or issues I was describing how to use `@Extensions()` to put custom metadata and middlewares to implement custom authorization middleware that fit your needs.

**Reply by emadbaqeri** [reply]
> In some discussion or issues I was describing how to use `@Extensions()` to put custom metadata and middlewares to implement custom authorization middleware that fit your needs.

Yeah I have also read that thread I think. But the current implementation of `@Authorized` decorator has enough flexibility and suits my needs after applying tricks to my own design.
I should mention that this is an awesome tool that you have built and I'm intending to contribute in my down time
